### ISSUE TITLE
Snowpark function `array_agg` converts data types to string

### ISSUE THREAD

**Issue by FlorianWilhelm** [snowflakedb/snowpark-python#1199]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

   Snowsight Python notebook with snowflake-snowpark-python 1.11.1

2. What operating system and processor architecture are you using?

   Snowsight Python notebook with snowflake-snowpark-python 1.11.1

3. What are the component versions in the environment (`pip freeze`)?

   Snowsight Python notebook with snowflake-snowpark-python 1.11.1

4. What did you do?

  I used the function `array_agg` and a column of big ints.

 ```python
import snowflake.snowpark as snowpark
from snowflake.snowpark.functions import col, array_agg

def main(session: snowpark.Session): 
    df = session.create_dataframe([[1], [2], [3], [1]], schema=["a"])
    print("Initial dtypes:", df.dtypes)
    df = df.select(array_agg("a", False).alias("result"))
    print("After array_agg dtypes:", df.dtypes) 
    return df
 ```

5. What did you expect to see?

   I hoped to get a single row and a column with dtype `array<bigint>`. Instead I got `array<string>`. 

  The PY output when runing the code above is:
   ```console
  Initial dtypes: [('A', 'bigint')]
  After array_agg dtypes: [('RESULT', 'array<string>')]
   ```

Why `array<string>`, this is crazy, but here is my number! as string? 🎤

**Comment by sfc-gh-yixie** [comment]
Yes we now use json string for arrays. A feature is under development for typed arrays.
